### PR TITLE
Update react-native-testing-library dependency references

### PR DESCRIPTION
**Pull request by luizbaldi**
Hi, thanks for creating such a useful repo :)

I just wanted to update the references for `react-native-testing-library` because it's now officially under the _"testing library umbrella"_, so the package name has changed and I also took the chance to make other subtle changes.

I'm happy to make any changes you find necessary.

### CODE REVIEW

**Comment by CodingItWrong** [comment]
Thanks! I will try to give it a run-through soon to ensure it all works locally for me as you've updated it, then I'll let you know if I have any other requests.

File: docs/component/testing.md
Comment (CodingItWrong) [inline docs/component/testing.md]: I think I'm actually going to update everywhere `react-native-testing-library` shows in this format to "React Native Testing Library", now that the package name is written out of order as `@testing-library/react-native`. No need for you to make that change in this PR; I'll make it in a followon.

File: docs/component/testing.md
Comment (CodingItWrong) [inline docs/component/testing.md:115]: Love it

File: docs/component/testing.md
Comment (CodingItWrong) [inline docs/component/testing.md]: I understand reasons why it might be preferable to set an expectation on the number of times called, but it's not the way I prefer to structure simple mock tests, and not related to the core of this PR. Could you revert this addition and the other one below?

**Review by CodingItWrong — CHANGES_REQUESTED** [review]
Package name updates work great—thank you for the update! Just a request to revert an unrelated change and then it should be set to merge.

File: docs/component/testing.md
Comment (luizbaldi) [reply docs/component/testing.md]: Ok.

File: docs/component/testing.md
Comment (luizbaldi) [reply docs/component/testing.md]: Sure.

**Comment by luizbaldi** [comment]
@CodingItWrong All good to be re-reviewed :)

File: docs/component/testing.md
Comment (CodingItWrong) [inline docs/component/testing.md]: There is still an addition here that needs to be removed as per my previous request. After that, will be set to merge.

Comment (luizbaldi) [reply docs/component/testing.md]: Ohh, sorry about that. I missed this one 😅

**Review by CodingItWrong — APPROVED** [review]
Thank you!

**Comment by CodingItWrong** [comment]
@luizbaldi Deployed: https://reactnativetesting.io/component/setup.html

**Pull request merged**
